keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md

```diff
@@ -99,10 +107,25 @@ The kube-controller-manager will still import the cloud provider implementations
 
 #### Phase 3 - Migrating Provider Code to Provider Repos
 
-In Phase 3, all code in `k8s.io/kubernetes/staging/src/k8s.io/legacy-cloud-providers/<provider>` will be removed and development of each cloud provider should be done in their respective external repos. It's important that by this phase, both in-tree and out-of-tree cloud providers are tested and production ready. Ideally most Kubernetes clusters in production should be using the out-of-tree provider before in-tree support is removed. A plan to migrate existing clusters from using the `kube-controller-manager` to the `cloud-controller-manager` is currently being developed. More details soon.
+In Phase 3, feature development is no longer accepted in `k8s.io/kubernetes/staging/src/k8s.io/legacy-cloud-providers/<provider>` and development of each cloud provider should be done in their respective external repos. It's important that by this phase, both in-tree and out-of-tree cloud providers are tested and production ready. Ideally most Kubernetes clusters in production should be using the out-of-tree provider before in-tree support is removed. A plan to migrate existing clusters from using the `kube-controller-manager` to the `cloud-controller-manager` is currently being developed. More details soon.
 
 External cloud providers can optionally still import providers from `k8s.io/legacy-cloud-providers` but no core components in `k8s.io/kubernetes` will import the legacy provider and the respective staging directory will be removed along with all its dependencies.
 
+#### Phase 4 - Disabling In-Tree Providers
+
+In Phase 4, two feature gates will be introduced to gradually disable and remove in-tree cloud providers:
```

Comment:
Worth mentioning that this does not cover either cloud provider storage solutions or credential providers? Also worth noting that by disable in-tree providers you may disable things like cloud based CIDR allocation which may not be immediately obvious? (Hence the need for testing)

Reply:
I would also emphasize the Beta part of this as I believe its the real justification. Cloud providers can always test by turning off the cloud provider flag themselves. So why bother adding the feature flag? The Beta flag roll out is a wake up call forcing them to see and acknowledge the coming changes.

Reply:
>  cloud based CIDR allocation which may not be immediately obvious

Actually, cloud CIDR allocator would be impacted because it depends on `--cloud-provider=gce`

Reply:
What is the behavior around flags such as SSHUser on the KAS which requires a supporting cloud provider?

Reply:
updated to include CSI part

Reply:
> What is the behavior around flags such as SSHUser on the KAS which requires a supporting cloud provider?

If the gate is enabled, and component has `--cloud-provider=gce` set, then component will exit. If any SSHUser depends on --cloud-provider being set, it will be disabled with the feature gate